src/components/popup.js

```diff
@@ -102,8 +104,25 @@ export default class Popup extends BaseControl {
     return anchor;
   }
 
+  /*
+   * Hack -
+   * React's `onClick` is called before mjolnir.js' `click` event (aka `tap` from hammer.js)
+   * which has a configurable delay.
+   * If we close the popup on the React event, by the time `click` fires, this component will
+   * have been unmounted, thus `captureClick` will not work.
+   * Instead, we flag the popup as closed on the React event, and actually close it on the hammer.js
+   * event.
+   */
+  _onClick(evt) {
```

Comment:
Nit: My sense is that this is a bit more than a `hack`, this is a key part of our "deep React" integration. Perhaps we could make slightly better provisions for it in the base class so that the code is less "hacky" here, if possible avoid the `super` call (which I personally think should be restricted to constructors if at all possible) and maybe a more clear function name to override?